src/Workspaces/Core/Desktop/Workspace/SQLite/SQLitePersistentStorage_StringIds.cs

```diff
@@ -12,23 +12,34 @@ internal partial class SQLitePersistentStorage
     {
         private readonly ConcurrentDictionary<string, int> _stringToIdMap = new ConcurrentDictionary<string, int>();
 
-        private void FetchStringTable(SqlConnection connection)
+        private bool TryFetchStringTable(SqlConnection connection)
         {
-            using (var resettableStatement = connection.GetResettableStatement(
-                $@"select * from ""{StringInfoTableName}"""))
+            try
             {
-                var statement = resettableStatement.Statement;
-                while (statement.Step() == Result.ROW)
+                using (var resettableStatement = connection.GetResettableStatement(
+                    $@"select * from ""{StringInfoTableName}"""))
                 {
-                    var id = statement.GetInt32At(columnIndex: 0);
-                    var value = statement.GetStringAt(columnIndex: 1);
-
-                    // Note that TryAdd won't overwrite an existing string->id pair.  That's what
-                    // we want.  we don't want the strings we've allocated from the DB to be what
-                    // we hold onto.  We'd rather hold onto the strings we get from sources like
-                    // the workspaces.  This helps avoid unnecessary string instance duplication.
-                    _stringToIdMap.TryAdd(value, id);
+                    var statement = resettableStatement.Statement;
+                    while (statement.Step() == Result.ROW)
+                    {
+                        var id = statement.GetInt32At(columnIndex: 0);
+                        var value = statement.GetStringAt(columnIndex: 1);
+
+                        // Note that TryAdd won't overwrite an existing string->id pair.  That's what
+                        // we want.  we don't want the strings we've allocated from the DB to be what
+                        // we hold onto.  We'd rather hold onto the strings we get from sources like
+                        // the workspaces.  This helps avoid unnecessary string instance duplication.
+                        _stringToIdMap.TryAdd(value, id);
+                    }
                 }
+
+                return true;
+            }
+            catch (SqlException e) when (e.Result == Result.BUSY || e.Result == Result.LOCKED)
```

Comment:
I'm a little nervous about this.  Do we need to make the change that causes these 'locked' errors to start happening.  What gain are we getting to warrant it?

My specific concern is that this now means that many more sql calls may have to be protected from locked errors, where we didn't have to worry about that before.  There may be more places that are missing that may end up leading to actual application errors in the wild.

Reply:
My understanding is the BUSY and LOCKED are similar, and the one getting reported depends on the current connection configuration. The original configuration produced BUSY errors (which we saw in Watson), but the change to use a shared and exclusive locks (required for follow-up work) changed a subset of those cases to return LOCKED instead.

Reply:
@CyrusNajmabadi I'd like to note that `busy` states are almost the same as `locked`, so the argument about unknown places in code that may fail are relevant to the `busy`.

Reply:
If they are effectively 1:1, then i'm ok with things.  My concern was that locked might show up in different places where 'busy' woudl not have.  If you guys are convinced that that's not an issue, then i'm def ok with this then :)  Thanks!